### PR TITLE
Add summarize function to combiner

### PR DESCRIPTION
**Pull request by chrisvittal**
This performs the operations described in the `NOTE` at the end of `vcf_combiner.py`

The only thing that I feel may be confusing is that we recompute the `DP` info fields as the sum of the `DP` entry fields. We do this as `densify` (currently a placeholder) may populate a previously missing entry.

### CODE REVIEW

File: hail/python/hail/experimental/vcf_combiner.py [hunk 1]
```diff
@@ -120,6 +121,23 @@ def min_rep(locus, ref, alt):
...
+# FIXME Placeholder
+def densify(mt):
+    return mt
+
+
+def summarize(mt):
+    mt = densify(mt)
+    return mt.annotate_rows(info=mt.info.annotate(
+        DP=hl.agg.sum(mt.entry.DP),  # some DPs may have been missing during earlier combining operations
+        BaseQRankSum=hl.median(hl.agg.collect(mt.entry.BaseQRankSum)),
```
Comment (chrisvittal) [inline hail/python/hail/experimental/vcf_combiner.py]: Is this the best way to take the median over the entries?

Comment (tpoterba) [reply hail/python/hail/experimental/vcf_combiner.py]: yes.

Comment (tpoterba) [reply hail/python/hail/experimental/vcf_combiner.py]: 😢

Comment (cseed) [reply hail/python/hail/experimental/vcf_combiner.py]: I was wrong.

File: hail/python/hail/experimental/vcf_combiner.py [hunk 2]
```diff
@@ -168,3 +186,46 @@ def min_rep(locus, ref, alt):
...
+# Column fields:
+#     's': str
+# ----------------------------------------
+# Row fields:
+#     'locus': locus<GRCh38>
+#     'alleles': array<str>
+#     'rsid': str
+#     'filters': set<str>
+#     'info': struct {
+#         DP: int64,
```
Comment (jigold) [inline hail/python/hail/experimental/vcf_combiner.py]: Is this needed? The schema is different than the one above. I.e. The entry fields are `mt.entry.*`.

Comment (chrisvittal) [reply hail/python/hail/experimental/vcf_combiner.py]: If you are talking about `info.DP` in general, it's probably not necessary, and we could save some cycles/bytes not computing it, except in `summarize`.

My intention was to always have a "best effort" sparse matrix table, where the fields are as close as possible to the correct values that would be present in a VCF for the same samples.

What is present in the comments here is the output of `mt.describe()` after running a combiner sample pipeline.

**Pull request merged**